docs/DeveloperGuide.md

```diff
@@ -133,6 +133,44 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Room Features
+
+The Room family of features consist of the following features: Add Room, Edit Room, List Rooms, Find Room and Delete Room. 
+
+#### The Room Class
+The Room class consists of 4 fields, each of which contain their own methods to verify their respective input. This allows for a low degree of coupling, and individual fields can change their input verification rules without affecting the other classes. Similarly, the Room class can expand to contain more fields without affecting existing fields too.
+
+Examples of verification functions in each of the fields include `RoomNumber#isValidRoomNumber()`, `RoomType#isValidRoomType()`, etc.
+
+![The Room Class](images/room/RoomClass.png)
+
+The `Room` objects are stored in a `UniqueRoomList` which is held by `AddressBook`.  
+
+##### Alternatives considered
+* The common paradigm of leaving fields such as RoomNumber, RoomType etc as member variables of the Room class was considered. We chose not to go down this path for 2 main reasons:
```

Comment:
This seems different from the ab3 alternative format provided. Maybe using the provided format would be better? Do let me know if I am mistaken.

The referenced link is here: https://se-education.org/addressbook-level3/DeveloperGuide.html#design-consideration

Reply:
You're right, I have updated it in the commit below.